draft-ietf-quic-transport.md

```diff
@@ -3285,8 +3292,10 @@ zero-length Destination Connection ID MUST treat receipt of a NEW_CONNECTION_ID
 frame as a connection error of type PROTOCOL_VIOLATION.
 
 Transmission errors, timeouts and retransmissions might cause the same
-NEW_CONNECTION_ID frame to be received multiple times. Receipt of the same
-frame multiple times MUST NOT be treated as a connection error.
+NEW_CONNECTION_ID frame to be received multiple times.  Receipt of the same
+frame multiple times MUST NOT be treated as a connection error.  A receiver can
+use the sequence number supplied in the NEW_CONNECTION_ID frame to identify new
+connection IDs from old ones.
```

Comment:
Q: What happens if the CID sequence number increases by a large amount(ie: 1000)?  If the same sequence number arrives with a different connection ID, is the receiver supposed to close the connection, ignore the second one?

Reply:
> I think we need to require the sender to always increase by one.

On the flip-side, does it require the sender to have a window of CIDs (i.e. the maximum number of CIDs in-flight)?

I ask this because unless we have such maximum, it is impossible for the receiver to enforce the rule.
If we have an enforcement, then the receiver could use a simple data structure (like holding the most-recent N CIDs where the recency is identified by the sequence number). But otherwise, the receivers need to be aware that the sender can try to consume large amount of state by sending a huge gap...